evm/multisig.js

```diff
@@ -0,0 +1,325 @@
+            }
+
+            if (!isNumber(parseFloat(withdrawAmount)) || parseFloat(withdrawAmount) === 0) {
+                throw new Error(`Invalid withdraw amount: ${withdrawAmount}`);
+            }
+
+            await preExecutionChecks(multisigContract, multisigAction, wallet, recipientAddress, '0x', withdrawAmount, yes);
+
+            const balance = await provider.getBalance(multisigContract.address);
+
+            if (balance.lt(withdrawAmount)) {
+                throw new Error(
+                    `Contract balance ${formatEther(BigNumber.from(balance))} is less than withdraw amount: ${formatEther(
+                        BigNumber.from(withdrawAmount),
+                    )}`,
+                );
+            }
+
+            try {
+                const tx = await multisigContract.withdraw(recipientAddress, withdrawAmount);
+                await tx.wait();
+            } catch (error) {
+                printError(error);
+            }
+
+            break;
+        }
+
+        case 'executeMultisigProposal': {
+            if (!isAddress(target) && target !== AddressZero) {
+                throw new Error(`Invalid target for execute multisig proposal: ${target}`);
+            }
+
+            if (!isValidCalldata(calldata)) {
+                throw new Error(`Invalid calldata for execute multisig proposal: ${calldata}`);
+            }
+
+            if (calldata === '0x' && !yes) {
+                printWarn(`Calldata for execute multisig proposal is empty.`);
+                const answer = readlineSync.question(`Proceed with ${multisigAction}?`);
+                if (answer !== 'y') return;
+            }
+
+            if (!isNumber(parseFloat(nativeValue))) {
+                throw new Error(`Invalid native value for execute multisig proposal: ${nativeValue}`);
+            }
+
+            const governance = chain.contracts.AxelarServiceGovernance?.address;
+
+            if (!isAddress(governance) && governance !== AddressZero) {
+                throw new Error(`Missing AxelarServiceGovernance address in the chain info.`);
+            }
+
+            const governanceContract = new Contract(governance, IGovernance.abi, wallet);
+
+            await preExecutionChecks(governanceContract, multisigAction, wallet, target, calldata, nativeValue, yes);
+
+            const balance = await provider.getBalance(governance);
+
+            if (balance.lt(nativeValue)) {
+                throw new Error(
+                    `AxelarServiceGovernance balance ${formatEther(
+                        BigNumber.from(balance),
+                    )} is less than native value amount: ${formatEther(BigNumber.from(nativeValue))}`,
+                );
+            }
+
+            try {
+                const tx = await governanceContract.executeMultisigProposal(target, calldata, nativeValue);
+                await tx.wait();
+            } catch (error) {
+                printError(error);
+            }
```

Comment:
don't think we need to catch the failure for execution. and if we want to allow that, it's simpler to do in `main`